lib/puppet/type/k5login.rb

```diff
@@ -31,10 +32,60 @@
     defaultto { "644" }
   end
 
+  # To manage the selinux user of the file
+  newproperty(:seluser) do
+    desc "What the SELinux user component of the context of the file should be.
+      Any valid SELinux user component is accepted.  For example `user_u`.
+      If not specified it defaults to the value returned by matchpathcon for
+      the file, if any exists.  Only valid on systems with SELinux support
+      enabled."
+
+    defaultto { self.retrieve_default_context(:seluser) }
```

Comment:
Oops, it looks like this is defined in

https://github.com/puppetlabs/puppet/blob/51636c065e0625b32af2cd98f2060eb7428704f5/lib/puppet/type/file/selcontext.rb#L40

I was assuming this was defined in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/selinux.rb instead.

Let's revert this back to the defaults you originally had so we can get this merged in. I've opened [PUP-8233](https://tickets.puppetlabs.com/browse/PUP-8233) to move `retrieve_default_context` to `util/selinux.rb` so we can use these defaults in both places in the future.

Reply:
Reverted and pushed new version